context.go

```diff
@@ -0,0 +1,23 @@
+package zipkin
+
+import (
+	"context"
+)
+
+// SpanFromContext retrieves a Zipkin Span from Go's context propagation
+// mechanism if found. If not found, returns nil.
+func SpanFromContext(ctx context.Context) Span {
+	if s, ok := ctx.Value(spanKey).(Span); ok {
+		return s
+	}
+	return nil
```

Comment:
Should we return a pointer here?

Reply:
Span is an interface (which is either nil or holds a pointer to an implementation of the Span interface). Thus this is the correct case... we have two span implementations of which one is a noop in case we have a situation in which either tracing is set to noop or noop spans for unsampled traces is requested

Reply:
To get to the broader discussion on if it should allow for a nil...I think yes... the alternative is having to deal with an empty implementation or additional error message. Neither is idiomatic in case of Go Context.

Reply:
As long as Span is an interface I am OK with it.